process explorer/backend/LocalCollector/Connections/ConnectionMonitor.cs

```diff
@@ -0,0 +1,38 @@
+﻿/* Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. */
+
+using LocalCollector.Connections;
+
+namespace ProcessExplorer.Entities.Connections
+{
+    public class ConnectionMonitor 
+    {
+        public ConnectionMonitorDto Data { get; set; } = new ConnectionMonitorDto();
+        public ConnectionMonitor()
+        {
+            Data.Connections = new SynchronizedCollection<ConnectionDto>();
+        }
+        public ConnectionMonitor(SynchronizedCollection<ConnectionDto> connections)
+            => this.Data.Connections = connections;
+        public void AddConnection(ConnectionDto connectionInfo)
+            => Data?.Connections?.Add(connectionInfo);
```

Comment:
The null-checks are inconsistent in this class. Either you should check for null everywhere if that's a valid state or if this.Data cannot be null then you should remove the null checks.